lib/stats-query-commands.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2017 Balabit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "stats-query-commands.h"
+#include "stats/stats-query.h"
+#include "messages.h"
+
+enum QueryCommands
+{
+  QUERY_GET = 0,
+  QUERY_GET_AGGREGATE,
+  QUERY_LIST,
+  QUERY_CMD_MAX
+};
+
+typedef GString *(*QUERY_CMD)(const gchar *hds_path);
+
+static GString *
+_query_get(const gchar *hds_path)
+{
+  return stats_query_get(hds_path);
+}
+
+static GString *
+_query_list(const gchar *hds_path)
+{
+  return stats_query_list(hds_path);
+}
+
+static GString *
+_query_get_aggregated(const gchar *hds_path)
+{
+  return stats_query_get_aggregated(hds_path);
+}
+
+static gint
+_command_str_to_id(const gchar *cmd)
+{
+  if (g_str_equal(cmd, "GET_AGGREGATED"))
+    return QUERY_GET_AGGREGATE;
+
+  if (g_str_equal(cmd, "GET"))
+    return QUERY_GET;
+
+  if (g_str_equal(cmd, "LIST"))
+    return QUERY_LIST;
+
+  msg_error("syslog-ng-ctl query got unknown command", evt_tag_str("command", cmd));
+
+  return QUERY_CMD_MAX;
+}
+
+static QUERY_CMD QUERY_CMDS[] =
+{
+  _query_get,
+  _query_get_aggregated,
+  _query_list
+};
+
```

Comment:
if we have an array like this, we might as well use to store the "string" representation of commands and use that instead of an if-else-if function to look up the command id.